static/panes/cfg-view.ts

```diff
@@ -0,0 +1,419 @@
+// Copyright (c) 2017, Najjar Chedy
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import vis from "vis";
+import _ from 'underscore';
+import { Toggles } from '../toggles';
+import { ga } from "../analytics";
+import TomSelect from 'tom-select';
+import { Container } from "golden-layout";
+import { CfgState } from "./cfg-view.interfaces";
+
+export class Cfg {
+    container: Container;
+    eventHub: any;
+    domRoot: JQuery;
+    defaultCfgOutput: object;
+    binaryModeSupport: object;
+    savedPos: any;
+    savedScale: any;
+    needsMove: boolean;
+    currentFunc: string;
+    functions: { [key: string]: any };
```

Comment:
This could be typed as `Record<string, any>`

Reply:
I was reading the documentation of [Index Signatures](https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures) and [Mapped Types](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html) and it seems to me that index signatures are the preferred way here because it's encoding a "dictionary".

Reply:
I definitely like this more than `Record<x, y>`, even though it's probably equally vague.